bot.go

```diff
@@ -63,11 +63,17 @@ func (bot *BotAPI) MakeRequest(endpoint string, params url.Values) (APIResponse,
 	method := fmt.Sprintf(APIEndpoint, bot.Token, endpoint)
 
 	resp, err := bot.Client.PostForm(method, params)
+
 	if err != nil {
 		return APIResponse{}, err
 	}
 	defer resp.Body.Close()
 
+	// Check http response status
```

Comment:
it was necessary to divide it into commits. 

We cannot simple reject all not OK response codes because telegram response with valid json in some cases, such as 401. This can repeat the usual description of HTTP errors, or it may not repeat
![2017-12-08-014234_1027x359_scrot](https://user-images.githubusercontent.com/10848952/33742433-5f639c7c-dbb9-11e7-862f-afca48f3d6d4.png)